app/src/main/java/nu/yona/app/ui/dashboard/PerDayFragment.java

```diff
@@ -94,16 +87,25 @@ public void onScrolled(RecyclerView recyclerView, int dx, int dy)
 		}
 	};
 
+	private void loadItemsOnScroll(int dy)
+	{
+		int visibleItemCount = mLayoutManager.getChildCount();
+		int totalItemCount = mLayoutManager.getItemCount();
+		int firstVisibleItemPosition = mLayoutManager.findFirstVisibleItemPosition();
+		if (dy > 0 && ((visibleItemCount + firstVisibleItemPosition) >= totalItemCount))
+		{
+			//load more items if number of items are less than the screen height or scroll view dy >0
+			loadMoreItems();
+		}
+	}
+
 	@Override
 	public void onCreate(@Nullable Bundle savedInstanceState)
 	{
 		super.onCreate(savedInstanceState);
-		if (getArguments().get(AppConstant.YONA_BUDDY_OBJ) != null)
+		if (getArguments().get(AppConstant.YONA_BUDDY_OBJ) != null && getArguments().get(AppConstant.YONA_BUDDY_OBJ) instanceof YonaBuddy)
```

Comment:
Good to combine the two if-statements, but even better: ``instanceof`` always returns false when the value is null, so this suffices:
```
		if (getArguments().get(AppConstant.YONA_BUDDY_OBJ) instanceof YonaBuddy)
```